src/lib/decentraland/sdk-components/cyclic-transform.ts

```diff
@@ -0,0 +1,59 @@
+import { SceneContext } from "../../babylon/scene/context"
+import { BabylonEntity } from "../../babylon/scene/entity"
+
+/**
+ * This function perform the scheduled parenting operations (scene.unparentedEntities)
+ * and prevents cycles in the final hierarchy.
+ * 
+ * If a cycle is found, then the parenting operation will remain pending until
+ * scene.hierarchyChanged is set to true again by a mutated TransformComponent.
```

Comment:
Great work on annotating this file! The comments really help, because they tell me _why_ things are happening, not only _how_ 🎉 I love it.

_Question_
What does "operation will remain pending" mean? Maybe that could be clarified further.